cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -154,6 +154,21 @@ impl PerCpuModeEncodings {
         self.enc64(inst.bind(I64), template.rex().w());
     }
 
+    /// Add encodings for `inst.i32` to X86_32.
+    /// Add an encoding for `inst.i32` to X86_64 that determines REX.RB at runtime.
+    /// Add encoding for `inst.i64` to X86_64 with a REX.W prefix.
+    fn enc_i32_i64_runtime_rex_rb(
+        &mut self,
+        inst: impl Into<InstSpec>,
+        template_hardcoded_rex: Template,
+        template_runtime_rex_rb: Template,
+    ) {
+        let inst: InstSpec = inst.into();
+        self.enc32(inst.bind(I32), template_hardcoded_rex.nonrex());
```

Comment:
The 32-bits version could use the runtime-decision too. With some guarding against the ISA's pointer size, it could be near to free in terms of runtime cost.